File: data/derived_cot/rq1_traces/translation/qwen/py_to_cpp/code_output/JSONProcessor.cpp
#include <nlohmann/json.hpp>
#include <fstream>
#include <filesystem>
#include <string>

using json = nlohmann::json;

class JSONProcessor {
public:
    int read_json(const std::string& file_path) {
        if (!std::filesystem::exists(file_path)) {
            return 0;
        }
        try {
            std::ifstream file(file_path);
            json data = json::parse(file);
            return data;
        } catch (...) {
            return -1;
        }
    }

    int write_json(const json& data, const std::string& file_path) {
        try {
            std::ofstream file(file_path);
            data.dump(file);
            return 1;
        } catch (...) {
            return -1;
        }
    }

    int process_json(const std::string& file_path, const std::string& remove_key) {
        auto data = read_json(file_path);
        if (data == 0 || data == -1) {
            return 0;
        }
        if (data.is_object() && data.find(remove_key) != data.end()) {
            data.erase(remove_key);
            write_json(data, file_path);
            return 1;
        }
        return 0;
    }
};

File: data/derived_cot/rq3_prompting/results/translation/qwen/java_to_py/code_output/JSONProcessor.py
import json
from pathlib import Path

class JSONProcessor:
    def read_json(self, filepath):
        if not Path(filepath).exists():
            return None
        try:
            with open(filepath, 'r') as file:
                return json.load(file)
        except Exception:
            return None

    def write_json(self, data, filepath):
        try:
            with open(filepath, 'w') as file:
                json.dump(data, file, indent=None)
            return True
        except Exception:
            return False

    def process_json(self, filepath, remove_key):
        data = self.read_json(filepath)
        if data is None:
            return False
        if remove_key in data:
            del data[remove_key]
            return self.write_json(data, filepath)
        return False